src/LowLevelMotionController.cpp

```diff
@@ -136,6 +142,16 @@ int main(int argc, char **argv)
 
     while (ros::ok())
     {
+        // Check the safety client before updating anything
+        ROS_ASSERT_MSG(!safety_client.isFatalActive(), "low_level_motion: fatal event from safety");
+
+        if(safety_client.isSafetyActive())
+        {
+            // Override whatever the Acceleration Planner wants to do and attempt to land
+            geometry_msgs::Twist twist;
+            twist.linear.z = -0.2; // Try to descend
+            quadController.setTargetVelocity(twist);
```

Comment:
I see, because the accelerationplanner is only updated in a callback.  It seems like this would be very easy to break in the future because it's dependent on ordering of statements that don't have an obvious need to be ordered that way.  I agree that the quadvelocitycontroller shouldn't be safety aware, but it might be useful to have a function in the quadvelocitycontroller that disables/enables the acceleration planner